content/blog/2025-07-17-unikraft-releases-v0.19.1.mdx

```diff
@@ -0,0 +1,83 @@
+
+### Changes in `0.19.1`
+
+* `lib/vfscore`: Harmonize handling of paths in \*at syscalls ([#1647](https://github.com/unikraft/unikraft/pull/1647)) by @andreittr
+* `lib/posix-process`: Fix clone() return value on thread create ([#1649](https://github.com/unikraft/unikraft/pull/1649)) by @michpappas
+* `lib/uksched`: Rework waitqueues to use thread data ([#1651](https://github.com/unikraft/unikraft/pull/1651)) by @andreittr
+* `lib/ukbitops`: Rework bit find ops for consistency ([#1652](https://github.com/unikraft/unikraft/pull/1652)) by @andreittr
+* `lib/ukboot`: Skip graceful shutdown on last child release ([#1653](https://github.com/unikraft/unikraft/pull/1653)) by @michpappas
+* `lib/ukrandom`: Fix undefined behavior ([#1655](https://github.com/unikraft/unikraft/pull/1655)) by @michpappas
+* `lib/posix-process`: Fix stack corruption in signal delivery ([#1657](https://github.com/unikraft/unikraft/pull/1657)) by @michpappas
+* `lib/vfscore`: Fix uninitialized mutex in dentry ([#1675](https://github.com/unikraft/unikraft/pull/1675)) by @andreittr
+
+:warning:  Compatibility notice:
+
+In `libukbitops`:
+
+* `uk_ffs()` / `uk_ffsl()` have been updated to return values consistent with libc's `ffs()` / `ffsl()` and `__builtin_ffs()` / `__builtin_ffsl()`.
+* Previous exact behavior of `uk_ffs()` / `uk_ffsl()` is now provided by `uk_lssb()` / `uk_lssbl()`.
+* `uk_fls()` / `uk_flsl()` are replaced by `uk_mssb()` / `uk_mssbl()`.
+
+### Catalog Updates
+
+Both the [`catalog`](https://github.com/unikraft/catalog) and [`catalog-core`](https://github.com/unikraft/catalog-core) repositories have received updates, primarily from the [Community and Development Lab](https://cdl.rosedu.org/) event, from [Google Summer of Code](https://summerofcode.withgoogle.com/) applicants and from [Unikraft Summer Workshop](https://unikraft.org/hackathons/usw25).
+Types of updates are:
+
+- Adding a new appplication to the catalog.
+- Updating the application version to the catalog.
+- Updating the naming schema.
+- Updating the unikernel runtime used.
+- Updating the configuration or documentation to have the application working with recent versions of Unikraft.
+
+### Community Activities
+
+#### Unikraft Summer Workshop 2025
+
+[Unikraft Summer Workshop 2025 (*USW'25*)](https://unikraft.org/hackathons/usw25) is a **free and virtual** workshop held by members of the Unikraft community.
+It focuses on cloud-native applications and on the unikernel technology that powers efficient and high performance cloud deployments.
+
+USW'25 took part for the weeks, between June 23 and July 12, 2025.
+It ended with a contributor hackathon on Saturday, July 12, 2025, 9am-5pm CEST.
+
+22 participants from all over the world took part in the event.
+In the end, they worked on 5 projects that resulted in approved contributions to Unikraft community repositories:
```

Comment:
```suggestion
In the end, they worked on 5 projects that resulted in contributions to Unikraft community repositories:
```

I think contributions is enough. Maybe "merged contributions", "approved" is a bit unclear.